File: flutter_app/lib/data/local/entity/action_entity.dart

import 'package:floor/floor.dart';
import 'package:flutter_app/data/remote/model/action_response.dart';

@Entity(tableName: 'action')
class ActionEntity {
  @primaryKey
  final String id;
  final String result;

  const ActionEntity({
    required this.id,
    required this.result,
  });

  factory ActionEntity.fromResponse(ActionResponse response) => ActionEntity(
    id: response.id,
    result: response.result,
  );
}

File: android_app/app/src/main/java/com/epam/kborodin/bored/data/local/entity/ActionEntity.kt
package com.epam.kborodin.bored.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "action")
data class ActionEntity(
    @PrimaryKey
    val id: String,
    @ColumnInfo(name = "result")
    val result: String
)